lib/rubocop/cop/rspec/dialect.rb

```diff
@@ -51,6 +51,12 @@ class Dialect < Base
         # @!method rspec_method?(node)
         def_node_matcher :rspec_method?, send_pattern('#ALL.all')
 
+        private_class_method def self.restrict_on_send
+          return [] unless RuboCop::RSpec::Language.config
+
```

Comment:
This is where it gets really interesting.

`restrict_on_send` implies that it would return a static value, while with dynamic, per-directory RSpec DSL conf, it won't.

`rubocop` runs cops in parallel, one per processor core.

Two issues here, one is somewhat suggested by the need fir this ﻿check

1 Is RuboCop caching this kist [here]https://github.com/rubocop/rubocop/blob/bc4ae3ef798c8db4f7427369777deb69be8f4d63/lib/rubocop/cop/commissioner.rb#L145)? If yes, supplying it dynamically is pointless

2 Is it safe to cache it on the class level in a multi-thread run?

Reply:
All of these were perspectives I had not considered, and I appreciate them very much. Thank you very much.
In this Pull Request, let's improve where we can optimize by setting `RESTRICT_ON_SEND`!